lib/mapping-transform.js

```diff
@@ -9,26 +9,17 @@ module.exports = class MappingTransform extends Transform {
     }
 
     _transform(chunk, _, callback) {
-        try {
-            Promise.resolve(this._function(chunk)).then(
-                this._handleResult(callback),
-                this._handleError.bind(this)
+        Promise.resolve(chunk)
+            .then(this._function)
+            .then(
+                (result) => callback(null, result),
+                (err) =>
+                    callback(
```

Comment:
need to handle cases where the callback throws. For example if the stream has already errors or closed.

I'm not sure there is anything to do with those errors but log and drop, but we should propagate an unhandled promise rejection.

Reply:
I was wondering the same. I can at least log the error for sure.

Reply:
I think callback would throw if the "downstream" marshaller errors, so that case will be handled by the streaming pipeline anyway. Logging is probably the best recourse.